sdk/tests/conformance2/uniforms/incompatible-texture-type-for-sampler.html

```diff
@@ -69,69 +69,71 @@
 {
   const t = textureInternalFormatInfo;
   // unsized formats
-  t[gl.ALPHA]              = { textureFormat: gl.ALPHA,           colorRenderable: true,  textureFilterable: true,  bytesPerElement: [1, 2, 2, 4],        type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT], };
-  t[gl.LUMINANCE]          = { textureFormat: gl.LUMINANCE,       colorRenderable: true,  textureFilterable: true,  bytesPerElement: [1, 2, 2, 4],        type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT], };
-  t[gl.LUMINANCE_ALPHA]    = { textureFormat: gl.LUMINANCE_ALPHA, colorRenderable: true,  textureFilterable: true,  bytesPerElement: [2, 4, 4, 8],        type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT], };
-  t[gl.RGB]                = { textureFormat: gl.RGB,             colorRenderable: true,  textureFilterable: true,  bytesPerElement: [3, 6, 6, 12, 2],    type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT, gl.UNSIGNED_SHORT_5_6_5], };
-  t[gl.RGBA]               = { textureFormat: gl.RGBA,            colorRenderable: true,  textureFilterable: true,  bytesPerElement: [4, 8, 8, 16, 2, 2], type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT, gl.UNSIGNED_SHORT_4_4_4_4, gl.UNSIGNED_SHORT_5_5_5_1], };
+  t[gl.ALPHA]              = { textureFormat: gl.ALPHA,            signed: false, int: true,  normalized: true,  depth: false, bytesPerElement: [1, 2, 2, 4],        type: [gl.UNSIGNED_BYTE, gl.HALF_FLOAT, gl.HALF_FLOAT_OES, gl.FLOAT], };
```

Comment:
So all the signed/int/normalized logic /works/, but fwiw this is what Gecko uses:
https://searchfox.org/mozilla-central/rev/adec563403271e78d1a057259b3e17fe557dfd91/dom/canvas/WebGLFormats.h#217
There's really only three categories here, so we might as well just label these FLOAT/INT/UNSIGNED_INT directly.

Reply:
Sorry, I was confused about normalized textures. I was thinking that they could be sampled as integer too, but they can't. This is much simpler now.